doc/development_guide/testing.rst

```diff
@@ -0,0 +1,131 @@
+.. -*- coding: utf-8 -*-
+.. _testing:
+
+==============
+ Testing
+==============
+
+.. _test_your_code:
+
+Test your code!
+----------------
+
+Pylint is very well tested and has a high code coverage. New contributions are not accepted
+unless they include tests.
+Pylint uses two types of tests: unittests and functional tests.
+
+  - The unittests can be found in the ``/pylint/test`` directory and they can
+    be used for testing almost anything Pylint related.
+
+  - The functional tests can be found in the ``/pylint/test/functional`` directory. They are
+    mainly used to test whether Pylint emits the correct messages.
+
+Before writing a new test it is often a good idea to ensure that your change isn't
+breaking a current test. You can run our tests using the tox_ package, as in::
+
+    python -m tox
+    python -m tox -epy36 # for Python 3.6 suite only
+    python -m tox -epylint # for running Pylint over Pylint's codebase
+    python -m tox -eformatting # for running formatting checks over Pylint's codebase
+
+It's usually a good idea to run tox_ with ``--recreate``. This flag tells tox_ to redownload
+all dependencies before running the tests. This can be important when a new version of
+astroid_ or any of the other dependencies has been published::
+
+    python -m tox --recreate # The entire tox environment will be recreated
+    python -m tox --recreate -e py310 # The python 3.10 tox environment will be recreated
+
+
+To run only a specific test suite, use a pattern for the test filename
+(**without** the ``.py`` extension), as in::
+
+    python -m tox -e py36 -- -k test_functional
+    python -m tox -e py36 -- -k  \*func\*
+    python -m tox --recreate -e py36 -- -k test_functional # With recreation of the environment
```

Comment:
```suggestion
    python -m tox -e py39 -- -k test_functional
    python -m tox -e py39 -- -k  \*func\*
    python -m tox --recreate -e py39 -- -k test_functional # With recreation of the environment
    
    # Faster alternative, but you must handle the dependencies with pip editable install
    pytest -k test_functional 
    pytest -k  \*func\* # Faster alternative
    
```
Also upgraded the version so we appear hype :)

Reply:
Should we add those last 3 lines? The 5 lines after this basically cover `pytest` already.